### DISCUSSION TITLE
RuntimeError: could not execute a primitive

### DISCUSSION THREAD

**Discussion by swap453** [PaddlePaddle/PaddleOCR#12623] [General]
File "/var/snap/reh/common/stitching/stitching_utils/get_set_brand_package.py", line 171, in setStitchedBestPackageType
    ocr_result = ocr.ocr(processed_img,cls=False)
  File "/snap/reh/2178/lib/python3.10/site-packages/paddleocr/paddleocr.py", line 668, in ocr
    dt_boxes, rec_res, _ = self.__call__(img, cls)
  File "/snap/reh/2178/lib/python3.10/site-packages/paddleocr/tools/infer/predict_system.py", line 105, in __call__
    rec_res, elapse = self.text_recognizer(img_crop_list)
  File "/snap/reh/2178/lib/python3.10/site-packages/paddleocr/tools/infer/predict_rec.py", line 616, in __call__
    self.predictor.run()
RuntimeError: could not execute a primitive

can someone help with this error ?

**Comment by GreatV** [comment]
paddlepaddle version?

**Reply by swap453** [reply]
2.6.1

**Comment by GreatV** [comment]
maybe you can refer: https://github.com/PaddlePaddle/PaddleOCR/issues/8607

**Reply by swap453** [reply]
Thanks I will take a look at it

**Comment by hardstone1998** [comment]
It's a problem with flask and paddlepaddle(cpu version).It can be solved using fastapi or paddlepaddle-gpu.Personally, it is recommended to replace flask with fastapi, because flask+paddlepaddle-gpu will cause the problem of gpu memory overflow.

**Reply by swap453** [reply]
@hardstone1998  where exactly we have to replace it with fastapi, can you please provide the code snippet

**Reply by hardstone1998** [reply]
This is my code demo. Fastapi can be used to run multiple times, but flask cannot

Use flask:
class TempPaddleOcr:
    def __init__(self):
        self.table_engine_en = PPStructure(recovery=True, lang='en', structure_version="PP-StructureV2",
                                           enable_mkldnn=False, use_gpu=True)
tempPaddleOcr = TempPaddleOcr()
@app.route('/image2text', methods=['POST'])
def image2text():
    file = request.files.get('file')
    filename = str(uuid.uuid1()) + "." + file.filename.split('.')[-1]
    if file:
        file.save(upload_folder + filename)
    img_path = upload_folder + filename
    img = cv2.imread(img_path)
    result = tempPaddleOcr.table_engine_en(img)
    return {'code': 200}
if __name__ == "__main__":
    print("服务被启动")
    app.run(host='0.0.0.0', port=8888)

Using fastapi:
class TempPaddleOcr:
    def __init__(self):
        self.table_engine_en = PPStructure(recovery=True, lang='en', structure_version="PP-StructureV2",
                                           enable_mkldnn=False, use_gpu=True)
tempPaddleOcr = TempPaddleOcr()
@app.post('/image2text')
async def image2text(file: UploadFile = File(...)):
    filename = str(uuid.uuid1()) + "." + file.filename.split('.')[-1]
    file_path = upload_folder + filename
    with open(file_path, "wb") as buffer:
        content = await file.read()
        buffer.write(content)
    img_path = upload_folder + filename
    img = cv2.imread(img_path)
    result = tempPaddleOcr.table_engine_en(img)
    return {'code': 200}
if __name__ == "__main__":
    print("服务被启动")
    uvicorn.run(app, host='0.0.0.0', port=8888)

**Reply by swap453** [reply]
Thank you. I will try that

**Comment by rttariverdi67** [comment]
the problem occures when you define ocr= PaddleOCR() once and use it multiple times.

you should use it in a way that when you wanna call ocr you shoud redefine it.

https://blog.csdn.net/with_me12138/article/details/136477210

**Comment by zhaxnb** [comment]
I used the "lang" parameter when using it, and this error did not occur. example:
original: table_engine = PPStructure(show_log=True) Error
now: table_engine = PPStructure(show_log=True, lang='ch') never Error

**Reply by wojiaoyishang** [reply]
it's amazing.
